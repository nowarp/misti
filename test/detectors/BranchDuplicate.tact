fun test1() {
    let a: Int = 42;
    if (a == 42) {
        a = 43; // bad
    } else {
        a = 43;
    }
}

fun test2(): Int {
    let a: Int = 42;
    return (a == 42) ? a + 43 : a + 43; // bad
}

fun test3() {
    let a: Int = 42;
    if (a == 42) {
        a = 44; // OK: No warning
    } else {
        a = 43;
    }
}

fun test4(): Int {
    let a: Int = 42;
    return (a == 42) ? a + 44 : a + 43; //OK: No warning
}

