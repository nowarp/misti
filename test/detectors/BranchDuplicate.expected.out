[HIGH] BranchDuplicate: Duplicated code in conditional branches is detected
test/detectors/BranchDuplicate.tact:3:5:
  2 |     let a: Int = 42;
> 3 |     if (a == 42) {
          ^
  4 |         a = 43; // bad
Help: Identical code in both branches detected. Refactor to eliminate redundancy.
See: https://nowarp.io/tools/misti/docs/detectors/BranchDuplicate

[HIGH] BranchDuplicate: Duplicated code in conditional branches is detected
test/detectors/BranchDuplicate.tact:12:12:
  11 |     let a: Int = 42;
> 12 |     return (a == 42) ? a + 43 : a + 43; // bad
                  ^
  13 | }
Help: Identical code in both branches detected. Refactor to eliminate redundancy.
See: https://nowarp.io/tools/misti/docs/detectors/BranchDuplicate